src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -331,7 +333,8 @@ private static void collectDeclarations(Deque<AbstractFrame> frameStack, DetailA
                 collectVariableDeclarations(ast, frame);
                 break;
             case TokenTypes.PARAMETER_DEF :
-                if (!CheckUtils.isReceiverParameter(ast)) {
```

Comment:
@ps-sp 
Judging by your UTs and your investigation, the patch is correct. The problem I faced was with the lambda parameters which were not enclosed in braces. You patch fixes the problem.